src/rpc/mining.cpp

```diff
@@ -961,3 +961,59 @@ UniValue getblocksubsidy(const UniValue& params, bool fHelp)
 
     return result;
 }
+
+UniValue getSidechainMerkleRoots(const UniValue& params, bool fHelp) {
+    if (fHelp || params.size() > 2)
+        throw runtime_error(
+                "getSidechainMerkleRoots transactions certificates\n"
+                "\nReturns Merkleroot and ScTxsCommitment.\n"
+                "\nArguments:\n"
+                "1. transactions         (array) Array of raw transactions (HEX format).\n"
+                "2. certificates         (array) Array of raw certificates (HEX format).\n"
+                "\nResult:\n"
+                "{\n"
+                "  \"merkleTree\" : \"xxx\"           (string) Merkleroot calculated on transactions and certificates.\n"
+                "  \"scTxsCommitment\" : \"xxxx\"      (string) scTxsCommitment calculated on certificates.\n"
+                "}\n"
+                "\nExamples:\n"
+                + HelpExampleCli("getSidechainMerkleRoots", "'[\"0100000001000000...\", ...]', '[\"0100000001000000...\", ...]'")
+                + HelpExampleRpc("getSidechainMerkleRoots", "'[\"0100000001000000...\", ...]', '[\"0100000001000000...\", ...]'")
+        );
+    LOCK(cs_main);
+
+    UniValue txsStr =  params[0].get_array();
```

Comment:
If the params are optional we should check params.size() before accessing params[0]